index.html

```diff
@@ -324,33 +323,34 @@
       <ol data-link-for="PerformanceObserverInit">
-        <li>Remove all unsupported names from the entryTypes sequence.</li>
-        <li>If the resulting '<a>entryTypes</a>' sequence is an empty sequence, abort
-          these steps. The user agent SHOULD notify developers when the steps are
-          aborted. A console warning listing ignored unsupported names in `entryTypes`
-          might be appropriate, for example.
-        </li>
-        <li>If the list of <a>registered performance observer</a> objects
-        associated with the <a>ECMAScript global environment</a> of the
-        interface object's contains a <a>registered performance observer</a>
-        that is the <a>context object</a>, replace the <a>context object</a>'s
-        `options` with <var>options</var>.
+        <li>Let <var>entry types</var> be <var>options</var> <a>entryTypes</a>
+          sequence.</li>
+        <li>Remove all unsupported types from <var>entry types</var>. The user
+          agent SHOULD notify developers if <var>entry types</var> is modified.
+          For example, a console warning listing removed types might be appropriate.</li>
+        <li>If the resulting <var>entry types</var> sequence is an empty sequence,
+          abort these steps. The user agent SHOULD notify developers when the
+          steps are aborted to notify that registration has been aborted. For
+          example, a console warning might be appropriate.</li>
+        <li>If the <a>list of registered performance observer objects</a> of
+          <a>current global object</a> contains a <a>registered performance
+          observer</a> that is the <a>context object</a>, replace its `options`,
+          with <var>options</var>.
         </li>
        <li>Otherwise, append a new <a>registered performance observer</a>
-        object to the list of <a>registered performance observer</a> objects
-        associated with the <a>ECMAScript global environment</a> of the
-        interface object, with the <a>context object</a> as `observer` and
-        <var>options</var> as the `options`.
+        object to the <a>list of registered performance observer objects</a> of
+        <a>current global object</a>, with the <a>context object</a> as
+        `observer` and <var>options</var> as the `options`.
         </li>
-        <li>If _options'_ <a for="PerformanceObserverInit">buffered</a> flag is
-          set, for each <dfn>entryType</dfn> of the `entryTypes` sequence:
+        <li>If <var>options</var>' <a>buffered</a> flag is set, for each
+          <var>entry type</var> in <var>entry types</var> sequence:
           <ol>
-            <li>Let <dfn>entries</dfn> be the <a>PerformanceEntryList</a>
+            <li>Let <var>entries</var> be the <a>PerformanceEntryList</a>
             object returned by the
             <a href="#filter-buffer-by-name-and-type"></a> algorithm with
             <var>buffer</var> set to <a>performance entry buffer</a>,
```

Comment:
"performance entry buffer" for which global?  This needs to be specified.  And tested.